src/lib.rs

```diff
@@ -452,243 +452,3 @@ mod tests {
         assert_eq!(x.find_gte(257), None);
     }
 }
-
-#[cfg(all(feature = "nightly", test))]
-mod b {
-    use super::OrderedCollection;
-    use std::collections::BTreeSet;
-    use test::black_box;
-    use test::Bencher;
-
-    // these benchmarks borrow from https://github.com/rust-lang/rust/pull/45333
-
-    enum Cache {
-        L1,
-        L2,
-        L3,
-    }
-
-    impl Cache {
-        pub fn size(&self) -> usize {
-            match *self {
-                Cache::L1 => 1000,      // 8kb
-                Cache::L2 => 10_000,    // 80kb
-                Cache::L3 => 1_000_000, // 8Mb
-            }
-        }
-    }
```

Comment:
I'm quite sad that we lose this for seeing how the performance scales with the varying cache sizes. Maybe we can highlight these points on the plot instead? Is there some other way we can represent this in Criterion? Perhaps by splitting the benchmark groups into three "ranges"?

Reply:
The reason I changed this, is because cache sizes vary widely from chip to chip, so giving the names 'L1', 'L2', 'L3' to some specific sizes is misleading.
Instead, the new code checks many different powers of two. 

Usually, the places of the 'bumps' in the graph you get from Criterion is very closely related to one of the cache sizes on that particular device. I'm not aware of any way to automatically draw extra lines on the resulting graphs, however.

Reply:
Yeah, that's a fair point and I'm sold :)